src/OrchardCore.Modules/OrchardCore.Media/SecureMediaPermissions.cs

```diff
@@ -3,12 +3,16 @@
 using System.Collections.ObjectModel;
 using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
 using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Cache;
 using OrchardCore.Media.Services;
+using OrchardCore.Modules;
 using OrchardCore.Security.Permissions;
 
 namespace OrchardCore.Media
 {
+    [Feature("OrchardCore.Media.Security")]
```

Comment:
Because it's defined in the same assembly as the media module itself, the `RoleUpdater` will not recognize secure media permissions as a separate feature. That causes the default permissions not being applied, when the feature is getting enabled.
Additionally, adding this attribute will group the new permissions under a separate category, which is beneficial as well IMO.

Reply:
To clarify, it's not actually the `RoleUpdater` but how the dependencies are mapped when the `ShellBlueprint` gets build...

Reply:
Hmm, interesting, this looks unintended. Could you please open a bug report?

Reply:
I created #15782.

Reply:
Note that I still want to keep the attribute, because I think the separate category in the permissions view is easier to work with.